src/core/dynamic-user.c

```diff
@@ -645,24 +646,22 @@ void dynamic_user_deserialize_one(Manager *m, const char *value, FDSet *fds) {
                 return;
         }
 
-        if ((fd0 = parse_fd(s0)) < 0 || !fdset_contains(fds, fd0)) {
-                log_debug("Unable to process dynamic user fd specification.");
+        fd0 = deserialize_fd(fds, s0);
+        if (fd0 < 0)
                 return;
-        }
 
-        if ((fd1 = parse_fd(s1)) < 0 || !fdset_contains(fds, fd1)) {
-                log_debug("Unable to process dynamic user fd specification.");
+        fd1 = deserialize_fd(fds, s1);
+        if (fd1 < 0)
                 return;
-        }
 
         r = dynamic_user_add(m, name, (int[]) { fd0, fd1 }, NULL);
```

Comment:
this is also different behaviour, the fd remained in the set if adding the user failed, now it's always removed

Reply:
effectively this makes no difference: if we leave it in the FDSet it will be flushed out at the end, as a left-over fd. Here we now close it early. Closed it will be anyway on failure.

It think this new approach is simpler, cleaner hence the change.

Reply:
as above